### PR TITLE
yosys: fix build on darwin

### PR DESCRIPTION
**Pull request by vbgl**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/yosys/default.nix [hunk 1]
```diff
@@ -32,7 +32,7 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
   nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ tcl readline libffi python3 bison flex ];
+  buildInputs = [ gcc tcl readline libffi python3 bison flex ];
```
Comment (LnL7) [inline pkgs/development/compilers/yosys/default.nix]: Why does this need gcc?

Comment (vbgl) [reply pkgs/development/compilers/yosys/default.nix]: I guess you should ask the authors…

There is a “config-clang” target in the makefile, but I did not manage to get the build go through (I got an error related to C++ header files). Should I investigate further or is the work-around of using GCC acceptable?

Comment (LnL7) [reply pkgs/development/compilers/yosys/default.nix]: Sorry for the delay. Just adding gcc doesn't work properly so I don't think it's actually needed. But if you don't want to spend more time on this, overriding the stdenv in all-packages.nix is the proper way to use gcc in all platforms.

```nix
{
  foo = callPackage ./foo { stdenv = gccStdenv; };
}
```

**Comment by vbgl** [comment]
Thanks for the tip, @LnL7. Meanwhile I figured out how to fix the build of `yosys` with `clang` without breaking the one of `abc`. There is thus no more need for `gcc`.

**Review by LnL7 — APPROVED** [review]

**Pull request merged**